usecases/auth/authorization/rbac/types.go

```diff
@@ -46,28 +46,21 @@ func policy(permission *models.Permission) (resource, verb, domain string) {
 	}
 
 	switch domain {
-	case rolesD:
-		resource = authorization.Roles()[0]
 	case cluster:
 		resource = authorization.Cluster()
+	case rolesD:
+		resource = authorization.Roles(*permission.Role)[0]
 	case collections:
-		if permission.Collection == nil {
-			resource = authorization.Collections("*")[0]
-		} else {
-			resource = authorization.Collections(*permission.Collection)[0]
-		}
+		resource = authorization.Collections(*permission.Collection)[0]
 	case tenants:
-		if permission.Collection == nil || permission.Tenant == nil {
-			resource = authorization.Shards("*", "*")[0]
-		} else {
-			resource = authorization.Shards(*permission.Collection, *permission.Tenant)[0]
-		}
+		resource = authorization.Shards(*permission.Collection, *permission.Tenant)[0]
 	case objects:
-		if permission.Collection == nil || permission.Tenant == nil || permission.Object == nil {
-			resource = authorization.Objects("*", "*", "*")
-		} else {
-			resource = authorization.Objects(*permission.Collection, *permission.Tenant, strfmt.UUID(*permission.Object))
-		}
+		resource = authorization.Objects(*permission.Collection, *permission.Tenant, strfmt.UUID(*permission.Object))
```

Comment:
IIRC the problem here was that permission.Collection can be `nil`

Reply:
yeah, i think we can ignore that part, i am gonna get it from Tommy's PR , it's better implemented there

Reply:
oki, so we wait for Tommys PR to be merged and then you rebase your changes?